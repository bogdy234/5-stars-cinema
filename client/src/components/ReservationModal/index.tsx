import { FC, ReactElement, useRef } from "react";
import Modal from "../Modal";
import CONSTANTS from "../../constants";
import Button from "../Button";
import useClickOutside from "../../utils/hooks/useClickOutside";

interface ReservationModalProps {
  showModal: boolean;
  onClickConfirm: () => void;
  message?: string;
  dateOptions: string[];
  timeOptions: string[];
  selectedDateOption: string;
  selectedTimeOption: string;
  onChangeDate: (date: string) => void;
  onChangeTime: (time: string) => void;
  hideModal: () => void;
}

const { CONFIRM, PLEASE_SELECT } = CONSTANTS.TEXT.RESERVATION_MODAL;

const ReservationModal: FC<ReservationModalProps> = ({
  showModal,
  onClickConfirm,
  dateOptions,
  timeOptions,
  selectedDateOption,
  selectedTimeOption,
  onChangeDate,
  onChangeTime,
  hideModal,
}: ReservationModalProps): ReactElement => {
  const ref = useRef(null);

  useClickOutside(ref, hideModal);
  return (
    <Modal showModal={showModal}>
      <div className={`flex justify-center items-center h-full`}>
        <div
          className={`w-5/6 md:w-1/2 lg:w-1/3 min-h-1/2 pb-10 rounded-3xl bg-gray-700`}
          ref={ref}
        >
          <div
            className={"flex justify-center flex-col items-center gap-4 mt-16"}
          >
            <div className={`text-2xl`}>{PLEASE_SELECT}</div>
            <select
              className={"bg-gray-300 w-1/2 h-14 rounded-xl pl-2 text-black"}
              value={selectedDateOption}
              onChange={(e) => onChangeDate(e.target.value)}
            >
              {dateOptions.map((option) => (
                <option key={`option-${Math.random() * 1000}`}>{option}</option>
              ))}
            </select>
          </div>
          <div className={"flex justify-center mt-10 text-black"}>
            <select
              className={"bg-gray-300 w-1/2 h-14 rounded-xl pl-2"}
              value={selectedTimeOption}
              onChange={(e) => onChangeTime(e.target.value)}
            >
              {timeOptions.map((option) => (
                <option key={`option-${Math.random() * 1000}`}>{option}</option>
              ))}
            </select>
          </div>
          <div className={`flex justify-center mt-20`}>
            <Button
              onClick={onClickConfirm}
              text={CONFIRM}
              className={`w-36 h-8 bg-primary text-black rounded hover:shadow-lg hover:shadow-cyan-300/50`}
            />
          </div>
        </div>
      </div>
    </Modal>
  );
};

export default ReservationModal;
