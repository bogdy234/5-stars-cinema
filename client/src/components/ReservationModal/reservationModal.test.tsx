import renderer from "react-test-renderer";
import ReservationModal from "./index";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <ReservationModal
        showModal={false}
        onClickConfirm={jest.fn()}
        dateOptions={[]}
        timeOptions={[]}
        selectedDateOption={""}
        selectedTimeOption={""}
        onChangeDate={jest.fn()}
        onChangeTime={jest.fn()}
        hideModal={jest.fn()}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
