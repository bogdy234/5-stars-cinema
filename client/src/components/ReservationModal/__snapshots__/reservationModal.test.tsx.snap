// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="fixed z-10 left-0 top-0 w-full h-full overflow-auto bg-black bg-opacity-50 hidden"
>
  <div
    className="flex justify-center items-center h-full"
  >
    <div
      className="w-5/6 md:w-1/2 lg:w-1/3 min-h-1/2 pb-10 rounded-3xl bg-gray-700"
    >
      <div
        className="flex justify-center flex-col items-center gap-4 mt-16"
      >
        <div
          className="text-2xl"
        >
          Selecteaza data si ora:
        </div>
        <select
          className="bg-gray-300 w-1/2 h-14 rounded-xl pl-2 text-black"
          onChange={[Function]}
          value=""
        />
      </div>
      <div
        className="flex justify-center mt-10 text-black"
      >
        <select
          className="bg-gray-300 w-1/2 h-14 rounded-xl pl-2"
          onChange={[Function]}
          value=""
        />
      </div>
      <div
        className="flex justify-center mt-20"
      >
        <button
          className="relative w-36 h-8 bg-primary text-black rounded hover:shadow-lg hover:shadow-cyan-300/50 undefined"
          onClick={[MockFunction]}
          type="button"
        >
          Confirma
        </button>
      </div>
    </div>
  </div>
</div>
`;
